language/stdlib/modules/transaction_fee_collection.mvir

```diff
@@ -0,0 +1,58 @@
+module TransactionFeeCollection {
+  import 0x0.LibraCoin;
+  import 0x0.Block;
+
+  // A pot of libra that is collected per transaction, and that will then be distributed to
+  // validators. This also tracks the current height of the blockchain in order to prevent double
+  // distributions per-block.
+  resource CollectedFees {
+    fees: LibraCoin.T,
+    current_height: u64,
+  }
+
+  public initialize() {
+    assert(get_txn_sender() == 0xA550C18, 0);
+    move_to_sender<CollectedFees>(CollectedFees {
+        fees: LibraCoin.zero(),
+        current_height: Block.get_current_height(),
+      });
+    return;
+  }
+
+  public pay_transaction_fee(fee_to_deposit: LibraCoin.T) acquires CollectedFees {
```

Comment:
Are we ok with someone calling this outside of the `prologue`? I think we should have a comment acknowledging that this is possible and arguing for why it's ok. 

Alternatively (and I actually think this is preferable) we should make `LibraAccount` depend on `TransactionFeeCollection` and make this private.

Reply:
It does indeed look a bit spooky. 

Adding a dependency and making it private doesn't solve too much? Unless you expect the VM to call it for you?

Perhaps we need `in_prologue()` and `in_epilogue()` builtins? (just a thought, I don't like the idea too much)

Reply:
This is a tricky one. The closest thing that would work is the `in_epilogue()` builtin if we want to restrict access. But, that seems hacky.

I think _at the moment_ leaving this open isn't a problem. However we should discuss with econ about this and any perverse incentives that it might create. But, as far as I can tell this wouldn't have any worse of an effect than being able to deposit money arbitrarily into accounts.